server/README.md

```diff
@@ -0,0 +1,181 @@
+
+`Attestation` message:
+```PROTO
+message AttestationRequest {
+    // Attestation evidence: https://github.com/confidential-containers/kbs/blob/main/docs/kbs_attestation_protocol.md#attestatio
+    bytes evidence = 1;
+    // Optional: Designate the user id. It should be keeped as "None" currently.
+    optional string user = 2;
+}
+```
+
+### Management
+
+It's mainly used to customize Attestation Server's configurations:
+- Customize each type of TEE's `Policy(.rego)` and `Reference Data(.json)` in order to evaluate it's TCB status precisely.
+    - GetPolicy
+    - GetReferenceData
+    - SetPolicy
+    - SetReferenceData
+    - ResetPolicy
+    - ResetReferenceData
+- Test/Evaluate the local `Policy(.rego)` or `Reference Data(.json)` with remote Attestation Server's OPA engine.
+    - TestOpa
+
+The supported types of TEE:
+```PROTO
+enum Tee {
+    Sgx = 0;
+    Tdx = 1;
+    SevSnp = 2;
+    Sample = 3;
+}
+```
+
+#### GetPolicy
+
+Get the specific TEE's OPA `Policy(.rego)`. The message:
+```PROTO
+message GetPolicyRequest {
+    // Supported TEE types
+    Tee tee = 1;
+    // Optional: Designate the user id. It should be keeped as "None" currently.
+    optional string user = 2;
+}
+```
+
+#### GetReferenceData
+
+Get the specific TEE's OPA `Reference Data(.json)`. The message:
+```PROTO
+message GetReferenceDataRequest {
+    // Supported TEE types
+    Tee tee = 1;
+    // Optional: Designate the user id. It should be keeped as "None" currently.
+    optional string user = 2;
+}
+```
+
+#### SetPolicy
+
+Set the specific TEE's OPA `Policy(.rego)`. It can make the Attestation Server evaluate the TEE TCB status according to specific user's preference. And it will echo error if the new `Policy(.rego)` syntax is illegal. The message:
+```PROTO
+message SetPolicyRequest {
+    // Supported TEE types
+    Tee tee = 1;
+    // Optional: Designate the user id. It should be keeped as "None" currently.
+    optional string user = 2;
+    // The "Policy(.rego)" file's content.
+    bytes content = 3;
+}
+```
+
+#### SetReferenceData
+
+Set the specific TEE's OPA `Reference Data(.json)`. It can attach a new released program's reference measurement value into the corresponding program's allow list. The message:
+```PROTO
+message SetReferenceDataRequest {
+    // Supported TEE types
+    Tee tee = 1;
+    // Optional: Designate the user id. It should be keeped as "None" currently.
+    optional string user = 2;
+    // The "Reference Data(.json)" file's content.
+    bytes content = 3;
+}
+```
+
+#### ResetPolicy
+
+Reset the specific TEE's OPA `Policy(.rego)` to default value. The message:
```

Comment:
It might make more sense to call this `DeletePolicy`. We aren't updating the policy to make it the same as the reference data, we are just removing the policy and then using the default policy instead. Same for reference data.

Reply:
Actually I also named this endpoint as `DeletePolicy`. But later I think `ResetPolicy` should be better. 
Because the `DeletePolicy` can only delete the user customized policy file, it can't delete the default policy file which is integrated inside AS Crate. After the customized policy file is deleted:
- The default policy will be selected and used during evidence attestation.
- `GetPolicy` will also return the default policy file.

Reply:
Maybe `RestoreDefaultPolicy`?

Reply:
I also renamed the `ResetReferenceData()` -> `RestoreDefaultReferenceData()`